R/derive_params_growth_age.R

```diff
@@ -262,25 +266,18 @@ derive_params_growth_age <- function(dataset,
       SD2pos = (M * (1 + 2 * L * S)^(1 / L)),
       SD3pos = (M * (1 + 3 * L * S)^(1 / L)),
       SD2neg = (M * (1 - 2 * L * S)^(1 / L)),
-      SD3neg = (M * (1 - 3 * L * S)^(1 / L))
+      SD3neg = (M * (1 - 3 * L * S)^(1 / L)),
+      age_bins = map(metadata_age, ~ set_bins(.x, breaks = bins$breaks, labels = bins$labels))
     )
 
   # Merge the dataset that contains the vs records and filter the L/M/S that fit the appropriate age
-  added_records <- dataset %>%
+  added_records <- dataset2 %>%
     filter(!!enexpr(parameter)) %>%
     left_join(.,
       processed_md,
-      by = c("sex_join", "ageu_join"),
-      relationship = "many-to-many"
+      by = c("sex_join", "ageu_join", "age_bins")
```

Comment:
we can use it as a unique join key to significantly reduce the run time of the appropriate join